augur/filter.py

```diff
@@ -3,381 +3,158 @@
-        print(error)
-        return 1
-
-
-    #####################################
-    #Filtering steps
-    #####################################
-
-    # remove sequences without meta data
-    tmp = [ ]
-    for seq_name in seq_keep:
-        if seq_name in meta_dict:
-            tmp.append(seq_name)
-        else:
-            print("No meta data for %s, excluding from all further analysis."%seq_name)
-    seq_keep = tmp
-
-    # remove strains explicitly excluded by name
-    # read list of strains to exclude from file and prune seq_keep
-    num_excluded_by_name = 0
-    if args.exclude:
-        try:
-            with open(args.exclude, 'r') as ifile:
-                to_exclude = set()
-                for line in ifile:
-                    if line[0] != comment_char:
-                        # strip whitespace and remove all text following comment character
-                        exclude_name = line.split(comment_char)[0].strip()
-                        to_exclude.add(exclude_name)
-            tmp = [seq_name for seq_name in seq_keep if seq_name not in to_exclude]
-            num_excluded_by_name = len(seq_keep) - len(tmp)
-            seq_keep = tmp
-        except FileNotFoundError as e:
-            print("ERROR: Could not open file of excluded strains '%s'" % args.exclude, file=sys.stderr)
-            sys.exit(1)
-
-    # exclude strain my metadata field like 'host=camel'
-    # match using lowercase
-    num_excluded_by_metadata = {}
-    if args.exclude_where:
-        for ex in args.exclude_where:
-            try:
-                col, val = re.split(r'!?=', ex)
-            except (ValueError,TypeError):
-                print("invalid --exclude-where clause \"%s\", should be of from property=value or property!=value"%ex)
-            else:
-                to_exclude = set()
-                for seq_name in seq_keep:
-                    if "!=" in ex: # i.e. property!=value requested
-                        if meta_dict[seq_name].get(col,'unknown').lower() != val.lower():
-                            to_exclude.add(seq_name)
-                    else: # i.e. property=value requested
-                        if meta_dict[seq_name].get(col,'unknown').lower() == val.lower():
-                            to_exclude.add(seq_name)
-                tmp = [seq_name for seq_name in seq_keep if seq_name not in to_exclude]
-                num_excluded_by_metadata[ex] = len(seq_keep) - len(tmp)
-                seq_keep = tmp
+    """
+    Filterer(args).run()
+
+
+class Filterer:
+    def __init__(self, args):
+        self.args = args
+        self.sequence_file = SequenceFile(args.sequences, args.metadata)
+
+        self.check_vcftools()
+
+        self.sequences = self.sequence_file.sequences
+
+    def run(self):
+        self.resulting_sequences # TODO have to run this here so the results are populated in each filter_obj, which print_summary() requires
```

Comment:
.resulting_sequences should probably be a function - this is a weird invocation.
maybe rename to `filter_sequences()` instead?

Reply:
Yeah, I was never happy with this. I'm pretty sure it was `filter_sequences()` in some intermediate incarnation, but it's probably better.